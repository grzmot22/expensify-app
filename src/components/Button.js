
import styled from "styled-components";
import config from "../styles/stylesConfig";

const Button = styled.button`
    color: white;
    background: ${props => props.background || config.COLORS.BLUE};
    border: none;
    display: inline-block;
    font-size: ${config.FONTS_SIZE.LARGE};
    font-weight: 300;
    line-height: 1;
    padding: ${config.SPACING.S_SIZE};
    text-decoration: none;   
`;

export default Button;