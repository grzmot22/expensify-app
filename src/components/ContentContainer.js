import styled from "styled-components";
import config from "../styles/stylesConfig";

const ContentContainer = styled.div`
    margin: 0 auto;
    padding: 0 ${config.SPACING.M_SIZE};
    max-width: 80rem;
`;

export default ContentContainer;