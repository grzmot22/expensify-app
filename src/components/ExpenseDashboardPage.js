import React from 'react';
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom';

const ExpenseDashboardPage = () => (
    <div>
    this is from my dashboard component
    </div>

);

export default ExpenseDashboardPage;