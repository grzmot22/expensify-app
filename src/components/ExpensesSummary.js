import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import styled from "styled-components";
import config from "../styles/stylesConfig";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {

  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');

  return (
    <PageHeader>
      <ContentContainer>
        <ContentContainerTitle>Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></ContentContainerTitle>
        <ContentContainerActions>
          <Button to="/create">Add Expense</Button>
        </ContentContainerActions>
      </ContentContainer>
    </PageHeader>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);

const PageHeader = styled.div`
    background: ${config.COLORS.OFF_WHITE};
    margin-bottom: ${config.SPACING.L_SIZE};
    padding: ${config.SPACING.L_SIZE} 0;
`;

const ContentContainer = styled.div`
    margin: 0 auto;
    padding: 0 ${config.SPACING.M_SIZE};
    max-width: 80rem;
`;

const ContentContainerTitle = styled.h1`
    font-weight: 300;
    margin: 0;
    span {
        font-weight: 700;
    }
`;

const ContentContainerActions = styled.div`
    margin-top: ${config.SPACING.M_SIZE};
`;

const Button = styled(Link)`

`;