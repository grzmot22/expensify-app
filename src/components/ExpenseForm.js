import React from 'react';
import moment from 'moment';
import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';
import styled from "styled-components";
import config from "../styles/stylesConfig";
import Button from "./Button";

export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        };
    }

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() =>({ description }));
    };

    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    };

    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));
        }
    }

    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({ createdAt }))
        }
    };

    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }))
    };

    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.amount) {
            this.setState(() =>({error: 'Please provide description and amount.'}))
        } else {
            this.setState(() =>({error: ''}))
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            });
        }
    }
    
    render() {
        return (        
            <FormContainer onSubmit={this.onSubmit}>
            {this.state.error && <FormError >{this.state.error}</FormError>}
            <TextInput
              type="text"
              placeholder="Description"
              autoFocus
              value={this.state.description}
              onChange={this.onDescriptionChange}
            />
            <TextInput
              type="text"
              placeholder="Amount"
              value={this.state.amount}
              onChange={this.onAmountChange}
            />
            <SingleDatePicker
              date={this.state.createdAt}
              onDateChange={this.onDateChange}
              focused={this.state.calendarFocused}
              onFocusChange={this.onFocusChange}
              numberOfMonths={1}
              isOutsideRange={() => false}
            />
            <TextArea
              placeholder="Add a note for your expense (optional)"
              value={this.state.note}
              onChange={this.onNoteChange}
            >
            </TextArea>
            <div>
              <Button>Save Expense</Button>
            </div>
          </FormContainer>
        )
    }
}

const FormContainer = styled.form`
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: ${config.SPACING.M_SIZE};
    }
`;

const FormError = styled.p`
    margin: 0 0 ${config.SPACING.M_SIZE} 0;
    font-style: italic;
`;


const TextInput = styled.input`
     border: 1px solid #cacccd;
    height: 50px;
    font-size: ${config.FONTS_SIZE.LARGE};
    font-weight: 300;
    padding: ${config.SPACING.S_SIZE};
`;

const TextArea = styled.textarea`
    border: 1px solid #cacccd;
    height: 10rem;
    font-size: ${config.FONTS_SIZE.LARGE};
    font-weight: 300;
    padding: ${config.SPACING.S_SIZE};
`;

