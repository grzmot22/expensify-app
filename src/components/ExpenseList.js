import React from 'react';
import { useSelector } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
import styled from 'styled-components';
import config from '../styles/stylesConfig';
import { darken } from 'polished';
import ContentContainer from './ContentContainer';

export const ExpenseList = () => {
  const expenses = useSelector((state) =>
    selectExpenses(state.expenses, state.filters),
  );

  return (
    <ContentContainer>
      <ListHeader>
        <HeaderMobile>Expenses</HeaderMobile>
        <HeaderDesktop>Expense</HeaderDesktop>
        <HeaderDesktop>Amount</HeaderDesktop>
      </ListHeader>
      <ListBody>
        {expenses.length === 0 ? (
          <ListMessage>
            <span>No expenses</span>
          </ListMessage>
        ) : (
          expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} />;
          })
        )}
      </ListBody>
    </ContentContainer>
  );
};

export default ExpenseList;

const ListHeader = styled.div`
  background: ${config.COLORS.OFF_WHITE};
  border: 1px solid ${darken(0.07, '#f7f7f7')};
  color: ${config.COLORS.GREY};
  display: flex;
  justify-content: space-between;
  padding: ${config.SPACING.S_SIZE} ${config.SPACING.M_SIZE};
`;

const ListBody = styled.div`
  margin-bottom: ${config.SPACING.M_SIZE};
  @media (min-width: ${config.SPACING.DESKTOP_BREAKPOINT}) {
    margin-bottom: ${config.SPACING.L_SIZE};
  }
`;

const ListMessage = styled.div`
  border: 1px solid ${darken(0.07, '#f7f7f7')};
  border-top: none;
  color: ${config.COLORS.DARK_GREY};
  display: flex;
  flex-direction: column;
  padding: ${config.SPACING.S_SIZE};
  text-decoration: none;
  transition: background 0.3s ease;

  @media (min-width: ${config.SPACING.DESKTOP_BREAKPOINT}) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    padding: ${config.SPACING.M_SIZE};
  }

  align-items: center;
  color: ${config.COLORS.GREY};
  justify-content: center;
  padding: ${config.SPACING.M_SIZE};
  &:hover {
    background: none;
  }
`;

const HeaderMobile = styled.div`
  @media (min-width: ${config.SPACING.DESKTOP_BREAKPOINT}) {
    display: none;
  }
`;

const HeaderDesktop = styled.div`
  @media (max-width: ${config.SPACING.DESKTOP_BREAKPOINT}) {
    display: none;
  }
`;
