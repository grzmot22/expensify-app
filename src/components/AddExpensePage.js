import React from 'react';
import { connect } from 'react-redux';
import { addExpense } from "../store/expenses/actions";
import ExpenseForm from './ExpenseForm';
import styled from "styled-components";
import config from "../styles/stylesConfig";
import ContentContainer from "./ContentContainer";

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense);
        this.props.history.push('/');
    };
    render() {
        return(
            <div>
                <PageHeader>
                    <ContentContainer>
                        <PageTitle>Add Expense</PageTitle>
                    </ContentContainer>
                </PageHeader>
                <ContentContainer>
                    <ExpenseForm 
                    onSubmit={this.onSubmit}
                    />
                </ContentContainer>
            </div>
        );
    }
};

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);

const PageHeader = styled.header`
    background: ${config.COLORS.OFF_WHITE};
    margin-bottom: ${config.SPACING.L_SIZE};
    padding: ${config.SPACING.L_SIZE} 0;
`;

const PageTitle = styled.h1`
    font-weight: 300;
    margin: 0;
    span {
        font-weight: 700;
    }
`;
