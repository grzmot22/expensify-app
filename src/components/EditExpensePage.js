import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from "../store/expenses/actions";
import styled from "styled-components";
import config from "../styles/stylesConfig";


export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id,expense);
        this.props.history.push('/');
    };

    onRemove = () => {
        this.props.startRemoveExpense({id: this.props.expense.id});
        this.props.history.push('/');
    };

    render() {
        return(
            <PageHeader>
                <ContentContainer>
                    <ContentContainer>
                        <PageTitle>Edit Expense</PageTitle>
                    </ContentContainer>
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                        />
                    <Button onClick={this.onRemove} >Remove Expense</Button>
                </ContentContainer>
           </PageHeader>
        )
    }

};

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(editExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(removeExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);

const PageHeader = styled.header`
    background: ${config.COLORS.OFF_WHITE};
    margin-bottom: ${config.SPACING.L_SIZE};
    padding: ${config.SPACING.L_SIZE} 0;
`;

const ContentContainer = styled.div`
    margin: 0 auto;
    padding: 0 ${config.SPACING.M_SIZE};
    max-width: 80rem;
`;

const PageTitle = styled.h1`
    font-weight: 300;
    margin: 0;
    span {
        font-weight: 700;
    }
`;

const Button = styled.button`
    color: white;
    background: #888888;
    border: none;
    display: inline-block;
    font-size: ${config.FONTS_SIZE.LARGE};
    font-weight: 300;
    line-height: 1;
    padding: ${config.SPACING.S_SIZE};
    text-decoration: none; 
`;
