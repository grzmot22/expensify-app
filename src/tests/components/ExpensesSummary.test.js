import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={234}/>);
    expect(wrapper).toMatchSnapshot();
})
test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={45} expensesTotal={3265432432}/>);
    expect(wrapper).toMatchSnapshot();
})

