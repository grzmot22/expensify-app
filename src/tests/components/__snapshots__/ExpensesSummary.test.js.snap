// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.34
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    45
     
    expenses
     totaling 
    $32,654,324.32
  </h1>
</div>
`;
